File: C++/Problems/Problems/problem_1221.cpp
#include "problem_1221.h"

bool problem_1221::test() {
    std::string s = "RLRRLLRLRL";

    int expected = 4;

    auto result = balancedStringSplit(s);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_1221::balancedStringSplit(std::string s) {
    int ls = 0;
    int rs = 0;
    int result = 0;
    for (auto c : s)
    {
        if (c == 'L')
            ls++;
        else
            rs++;
        if (ls == rs)
        {
            result++;
            ls = 0;
            rs = 0;
        }
    }

    return result;
}

File: Python/Problems/Problems/problem1221.py
from utils import print1DArray, print2DArray
from typing import List
from collections import Counter

from .problem import Problem


class Problem1221(Problem):
    def test(self) -> bool:
        s = "RLRRLLRLRL"

        expected = 4

        result = self.balancedStringSplit(s)

        print(result)

        return result == expected

    def balancedStringSplit(self, s: str) -> int:
        ls = 0
        rs = 0
        result = 0
        for c in s:
            if c == 'L':
                ls += 1
            else:
                rs += 1
            if ls == rs:
                result += 1
                ls = 0
                rs = 0

        return result